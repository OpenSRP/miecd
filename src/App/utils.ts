import { DEFAULT_PLAN_ID, KEYCLOAK_API_BASE_URL, KEYCLOAK_USERS_PAGE_SIZE, OPENSRP_API_BASE_URL } from 'configs/env';
import { URL_LOCATION_UNIT } from '../constants';

const openSrpRestBase = OPENSRP_API_BASE_URL + '/rest/';

export const baseProps = {
    baseURL: openSrpRestBase,
    opensrpBaseURL: openSrpRestBase,
    keycloakBaseURL: KEYCLOAK_API_BASE_URL,
};

export const locationUnitProps = {
    ...baseProps,
    filterByParentId: false,
};

export const newLocationUnitProps = {
    ...locationUnitProps,
    successURLGenerator: () => URL_LOCATION_UNIT,
    cancelURLGenerator: () => URL_LOCATION_UNIT,
    hidden: ['serviceType', 'latitude', 'longitude'],
};

export const editLocationProps = {
    ...newLocationUnitProps,
    ...locationUnitProps,
};

export const usersListProps = {
    ...baseProps,
    usersPageSize: KEYCLOAK_USERS_PAGE_SIZE,
};

export const teamAssignmentProps = {
    ...baseProps,
    defaultPlanId: DEFAULT_PLAN_ID,
};

export const createEditUserProps = {
    ...baseProps,
    userFormHidden: [],
};
