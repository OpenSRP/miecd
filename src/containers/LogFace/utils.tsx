import { replace, split, trim } from 'lodash';
import React from 'react';

/** convert the smsData message field from prose to more easily readable
 * point format
 *
 * @param message - the message prose.
 */
export const parseMessage = (message: string) => {
    const propValues = split(message, '\n');
    const replacedEquals = propValues.map(trim).map((entry) => replace(entry, / =\s*/, ' : '));
    return (
        <ul>
            {replacedEquals.map((value, index) => {
                return <li key={`${value}-${index}`}>{value}</li>;
            })}
        </ul>
    );
};
