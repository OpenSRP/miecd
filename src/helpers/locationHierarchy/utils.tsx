import { ParsedHierarchyNode, RawHierarchyNode, RawHierarchyNodeMap, RawOpenSRPHierarchy, TreeNode } from './types';
import { cloneDeep } from 'lodash';
import cycle from 'cycle';
import TreeModel from 'tree-model';

/** Parse the raw child hierarchy node map
 *
 * @param {RawHierarchyNodeMap} rawNodeMap - Object of raw hierarchy nodes
 * @param {string} parent - node parent id
 * @returns {Array<ParsedHierarchyNode>} Array of Parsed hierarchy nodes
 */
const parseChildren = (rawNodeMap: RawHierarchyNodeMap, parent: string) => {
    // skip key when destructuring
    const rawHierarchyNode: RawHierarchyNode[] = Object.entries(rawNodeMap).map(([, value]) => value);
    return rawHierarchyNode.map((child) => {
        const parsedNode: ParsedHierarchyNode = {
            ...child,
            title: child.label,
            key: `${child.id}-${parent}`,
            children: child.children ? parseChildren(child.children, parent) : [],
        };
        return parsedNode;
    });
};

/** parses the raw opensrp hierarchy to a hierarchy that we can quickly build
 * our tree model from.
 *
 * @param {RawOpenSRPHierarchy} raw - the response we get from opensrp
 * @returns {ParsedHierarchyNode} - returns Parent node with its children
 */
const parseHierarchy = (raw: RawOpenSRPHierarchy) => {
    // clone the locationTree, we are going to be mutating a copy
    const rawClone: RawOpenSRPHierarchy = cloneDeep(raw);

    // !IMPORTANT ASSUMPTION : locationsTreeClone has a single object under map, i.e there is only one root jurisdiction
    const { map } = rawClone.locationsHierarchy;
    // !IMPORTANT ASSUMPTION : locationsTreeClone has a single object under map, i.e there is only one root jurisdiction
    // skip key when destructuring
    const rawNode: RawHierarchyNode = Object.entries(map).map(([, value]) => value)[0];
    const parsedNode: ParsedHierarchyNode = {
        ...rawNode,
        title: rawNode.label,
        key: rawNode.id,
        children: rawNode.children ? parseChildren(rawNode.children, rawNode.id) : [],
    };

    return parsedNode;
};

/** takes the raw opensrp hierarchy response and creates a tree model structure
 *
 * @param {RawOpenSRPHierarchy} apiResponse - the response we get from opensrp
 * @returns {TreeNode} - returns root node
 */
export const generateJurisdictionTree = (apiResponse: RawOpenSRPHierarchy): TreeNode => {
    const tree = new TreeModel();
    const hierarchy = parseHierarchy(apiResponse);
    const root = tree.parse<ParsedHierarchyNode>(hierarchy);
    return root;
};

/**
 * serialize tree due to circular dependencies
 *
 * @param trees - trees to be serialized
 */
export const serializeTree = (trees: (TreeNode | undefined)[]) => {
    return JSON.stringify(trees.map((tree) => JSON.stringify(cycle.decycle(tree))));
};
