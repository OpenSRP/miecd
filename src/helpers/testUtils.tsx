import { mount, MountRendererProps } from 'enzyme';
import React from 'react';
import { I18nextProvider } from 'react-i18next';
import i18n from '../mls';
import { act } from 'react-dom/test-utils';

export const mountWithTranslations: typeof mount = (component: React.ReactNode, options?: MountRendererProps) => {
    return mount(<I18nextProvider i18n={i18n}>{component}</I18nextProvider>, options);
};

/**
 * Utility to await promises to resolve in tests
 * @param ms time in ms
 */
export async function waitForPromises() {
    await act(async () => {
        await new Promise((res) => setImmediate(res));
    });
}
