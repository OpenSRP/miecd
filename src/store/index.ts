import { connectReducer, getConnectedStore } from '@onaio/connected-reducer-registry';
import { gateKeeperReducer, gateKeeperReducerName } from '@onaio/gatekeeper';
import reducerRegistry, { combine } from '@onaio/redux-reducer-registry';
import session, { reducerName as sessionReducer } from '@onaio/session-reducer';
import { Dictionary } from '@onaio/utils';

/** Initial reducers in the reducer registry */
const defaultReducers: Dictionary = {
    router: connectReducer /** Dirty hack because connectRouter LocationChangeAction does not extend Redux.AnyAction */,
};

/** Add users reducer to registry */
defaultReducers[sessionReducer] = session;
defaultReducers[gateKeeperReducerName] = gateKeeperReducer;

/** The initial store for the reveal web app */
const store = getConnectedStore(defaultReducers);

/** Set listener to add reducers to store when registered */
reducerRegistry.setChangeListener((reducers) => {
    store.replaceReducer(combine(reducers));
});

export default store;
