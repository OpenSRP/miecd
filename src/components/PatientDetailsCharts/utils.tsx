import NoRecord from 'components/NoRecord';
import Ripple from 'components/page/Loading';
import { TFunction, useTranslation } from 'react-i18next';
import supersetFetch from 'services/superset';
import { getMonthNames } from '../../constants';
import { sortByEventDate } from '../../helpers/utils';
import { ChildChartData, MotherChartData } from 'store/ducks/chartData';
import superset, { SupersetFormData } from '@onaio/superset-connector';
import { ActionCreator } from 'redux';
import { GET_FORM_DATA_ROW_LIMIT } from 'configs/env';
import React from 'react';

/** factory for form data filters for chart pages */
export const chartFormDataFilters = (patientId: string) =>
    superset.getFormData(GET_FORM_DATA_ROW_LIMIT, [
        {
            comparator: patientId,
            operator: '==',
            subject: 'patient_id',
        },
    ]);

/** abstracts code that actually makes the superset Call since it is quite similar
 * @param supersetSlice - slice string
 * @param actionCreator - redux action creator
 * @param supersetService - the supersetService
 * @param supersetOptions - adhoc filters for superset call
 */
export async function chartSupersetCall<TAction, TResponse>(
    supersetSlice: string,
    actionCreator: ActionCreator<TAction>,
    supersetService: typeof supersetFetch = supersetFetch,
    supersetOptions: SupersetFormData | null = null,
) {
    const asyncOperation = supersetOptions
        ? supersetService(supersetSlice, supersetOptions)
        : supersetService(supersetSlice);
    return asyncOperation
        .then((result: TResponse[] | undefined) => {
            if (result) {
                actionCreator(result);
            }
        })
        .catch((error) => {
            throw error;
        });
}

/** props for BrokenLoading chart */
export interface BrokenLoadingChartPageProps {
    error?: Error;
    loading: boolean;
    broken: boolean;
}

/** dry component for showing loader or broken page */
export const BrokenLoadingChartPage = ({ error, loading, broken }: BrokenLoadingChartPageProps) => {
    const { t } = useTranslation();
    if (loading) {
        return <Ripple></Ripple>;
    }

    if (broken) {
        return <NoRecord message={t(`Chart could not load, an error occurred. ${error?.message ?? ''}`)} />;
    }
    return null;
};

/** create a chart data series containing the weight and height data series from smsEvents */
export const getWeightHeightDataSeries = (smsEvents: ChildChartData[], t: TFunction) => {
    const categories: string[] = [];
    const dataSeries: { data: (number | undefined)[]; name: string }[] = [
        { data: [], name: 'weight' },
        { data: [], name: 'height' },
    ];
    // make sure events are sorted from oldest to latest
    const sortedSmsEvents = sortByEventDate(smsEvents).reverse();

    for (const data of sortedSmsEvents) {
        const calendarCategoryObj = {
            month: new Date(data.event_date).getMonth(),
            year: new Date(data.event_date).getFullYear(),
        };
        if (data.weight && data.height) {
            categories.push(`${getMonthNames(t)[calendarCategoryObj.month]}/${calendarCategoryObj.year}`);
            dataSeries[0].data.push(data.weight);
            dataSeries[1].data.push(data.height);
        }
    }
    return {
        categories,
        dataSeries,
    };
};

/** create a chart data series containing the weight data series from smsEvents */
export const getWeightDataSeries = (smsEvents: MotherChartData[], t: TFunction) => {
    const categories: string[] = [];
    const dataSeries: { data: (number | undefined)[]; name: string }[] = [{ data: [], name: 'weight' }];
    // make sure events are sorted from oldest to latest
    const sortedSmsEvents = sortByEventDate(smsEvents).reverse();

    for (const data of sortedSmsEvents) {
        const calendarCategoryObj = {
            month: new Date(data.event_date).getMonth(),
            year: new Date(data.event_date).getFullYear(),
        };
        if (data.weight) {
            categories.push(`${getMonthNames(t)[calendarCategoryObj.month]}/${calendarCategoryObj.year}`);
            dataSeries[0].data.push(data.weight);
        }
    }
    return {
        categories,
        dataSeries,
    };
};

/** create a data series containing the diastolic and systolic data series from smsEvents */
export const getBloodPSeriesForChart = (smsEvents: MotherChartData[], t: TFunction) => {
    const categories: string[] = [];
    const dataSeries: { data: (number | undefined)[]; name: string }[] = [
        { data: [], name: 'systolic' },
        { data: [], name: 'diastolic' },
    ];
    // make sure events are sorted from oldest to latest
    const sortedSmsEvents = sortByEventDate(smsEvents).reverse();

    const regex = /\d{2,3}\S{0,3}\d{2,3}/;
    const singleBpValueRegex = /\d{2,3}/g;

    for (const data of sortedSmsEvents) {
        const calendarCategoryObj = {
            month: new Date(data.event_date).getMonth(),
            year: new Date(data.event_date).getFullYear(),
        };
        const hasValidBp = regex.test(data.bp);
        if (!hasValidBp) {
            continue;
        }
        categories.push(`${getMonthNames(t)[calendarCategoryObj.month]}/${calendarCategoryObj.year}`);
        const matchedValues = [...Array.from(data.bp.matchAll(singleBpValueRegex))];

        dataSeries[0].data.push(Number(matchedValues[0]));
        dataSeries[1].data.push(Number(matchedValues[1]));
    }
    return {
        categories,
        dataSeries,
    };
};
