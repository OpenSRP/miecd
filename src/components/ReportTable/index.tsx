import ListView from '@onaio/list-view';
import NoRecord from 'components/NoRecord';
import { keyBy, uniqWith } from 'lodash';
import React, { useState } from 'react';
import { TFunction, Trans, useTranslation } from 'react-i18next';
import Select from 'react-select';
import { Row } from 'reactstrap';
import {
    ANC_REPORT,
    BIRTH_REPORT,
    GESTATION_PERIOD,
    getMonthNames,
    NUTRITION_REGISTRATION,
    NUTRITION_REPORT,
    PREGNANCY_REGISTRATION,
} from '../../constants';
import { sortByEventDate } from '../../helpers/utils';
import { SmsData } from '../../store/ducks/sms_events';
import { Chart } from '../WeightAndHeightChart';
import './index.css';

interface ReportTableProps {
    singlePatientEvents: SmsData[];
    isChild: boolean;
}

const defaultProps = {
    singlePatientEvents: [],
    isChild: false,
};

export type ReportTableTypes = ReportTableProps;

function ReportTable(props: ReportTableTypes) {
    const { singlePatientEvents, isChild } = props;
    const smsTypesOfInterest = filterEventsByType(singlePatientEvents, isChild);
    let smsChunks = [smsTypesOfInterest];
    if (!isChild) {
        smsChunks = chunkByGravida(smsTypesOfInterest);
    }
    const [currentSmsChunkIndex, setCurrentSmsChunkIndex] = useState(smsChunks.length - 1);
    const { t } = useTranslation();

    const listViewProps = {
        data:
            smsChunks[currentSmsChunkIndex]?.map((smsData) => [
                smsData.sms_type,
                smsData.EventDate,
                smsData.health_worker_name,
                smsData.message,
            ]) ?? [],
        headerItems: [t('Report'), t('Date'), t('Reporter'), t('Message')],
        tableClass: 'table-container',
        tbodyClass: 'body',
        thClass: 'report-table__td',
        tdClass: 'report-table__td',
    };

    const currentSmsDataChunk = smsChunks[currentSmsChunkIndex];
    const filterOptions = pregnancyOptionsFilter(smsChunks, t);
    const filterValue = keyBy(filterOptions, (x) => x.value)[currentSmsChunkIndex];

    return (
        <>
            <div id="filter-panel">
                {!isChild && (
                    <>
                        <p>
                            <Trans t={t}>Showing reports for:&emsp;</Trans>
                        </p>
                        <div className="filters">
                            <Select
                                placeholder={t('Select pregnancy')}
                                options={filterOptions}
                                value={filterValue}
                                onChange={(val) => {
                                    setCurrentSmsChunkIndex(val?.value ?? smsChunks.length - 1);
                                }}
                                classNamePrefix="patient-details-filters"
                                isClearable={true}
                                id="pregnancy-filter"
                            />
                        </div>
                    </>
                )}
            </div>
            <Row id="tableRow">
                <ListView {...listViewProps} />
            </Row>
            {currentSmsDataChunk.length > 0 ? (
                <div id="chart">
                    {isChild ? (
                        <Chart
                            dataArray={getWeightHeightDataSeries(currentSmsDataChunk, t)}
                            chartWrapperId="child-nutrition-chart-1"
                            title={t('Weight Height Monitoring')}
                            legendString={t('Weight Height')}
                            units={t('cm')}
                            yAxisLabel={t('weight and height')}
                        />
                    ) : (
                        <>
                            <Chart
                                dataArray={getWeightDataSeries(currentSmsDataChunk, t)}
                                chartWrapperId="weight-chart-1"
                                title={t('Wight Monitoring')}
                                legendString={t('Weight')}
                                units={t('kg')}
                                yAxisLabel={t('weight')}
                            />

                            <Chart
                                dataArray={getBloodPSeriesForChart(currentSmsDataChunk, t)}
                                chartWrapperId="blood-pressure"
                                title="Blood Pressure"
                                legendString={t('Blood pressure')}
                                units=""
                                yAxisLabel={t('Blood Pressure')}
                            />
                        </>
                    )}
                </div>
            ) : (
                <NoRecord message={t('No data found')} />
            )}
        </>
    );
}

ReportTable.defaultProps = defaultProps;

export default ReportTable;

/** filter out sms events that this component need not concern itself with */
export const filterEventsByType = (smsData: SmsData[], isChild: boolean) => {
    const lowerCase = (x: string) => x.toLowerCase();
    const pregnancySmsEventTypes = [BIRTH_REPORT, ANC_REPORT, PREGNANCY_REGISTRATION].map(lowerCase);
    const nutritionSmsEventTypes = [NUTRITION_REPORT, NUTRITION_REGISTRATION].map(lowerCase);
    const smsTypesToUse = isChild ? nutritionSmsEventTypes : pregnancySmsEventTypes;

    return smsData.filter((sms) => {
        return smsTypesToUse.includes(lowerCase(sms.sms_type));
    });
};

/** helps chunk and group smsEvents by the pregnancies that they belong to */
export const chunkByGravida = (smsEvents: SmsData[]) => {
    const smsChunks: SmsData[][] = [];
    let chunkIndex = 0;
    if (!smsEvents.length) {
        return [[]];
    }

    smsEvents.forEach((event, idx) => {
        if (smsChunks[chunkIndex]) {
            if (
                event.sms_type === PREGNANCY_REGISTRATION ||
                (smsChunks[chunkIndex][idx - 1] && // check if same pregnancy based on the gestation period difference.
                    GESTATION_PERIOD <
                        Math.abs(Date.parse(event.EventDate) - Date.parse(smsChunks[chunkIndex][idx - 1].EventDate)))
            ) {
                chunkIndex += 1;
                if (!smsChunks[chunkIndex]) {
                    smsChunks[chunkIndex] = [];
                }
                smsChunks[chunkIndex].push(event);
            } else {
                smsChunks[chunkIndex].push(event);
            }
        } else {
            smsChunks[chunkIndex] = [];
            smsChunks[chunkIndex].push(event);
        }
    });

    // remove duplicate pregnancy registration chunks with respect to the gestation period
    return uniqWith(smsChunks, (chunk1, chunk2) => {
        const pregnancyRegTypeSms1 = chunk1.filter((sms) => sms.sms_type === PREGNANCY_REGISTRATION);
        const pregnancyRegTypeSms2 = chunk2.filter((sms) => sms.sms_type === PREGNANCY_REGISTRATION);
        if (pregnancyRegTypeSms1.length && pregnancyRegTypeSms2.length) {
            return (
                GESTATION_PERIOD >
                Math.abs(Date.parse(pregnancyRegTypeSms1[0].EventDate) - Date.parse(pregnancyRegTypeSms1[0].EventDate))
            );
        }
        return false;
    });
};

/** create a chart data series containing the weight and height data series from smsEvents */
export const getWeightHeightDataSeries = (smsEvents: SmsData[], t: TFunction) => {
    const categories: string[] = [];
    const dataSeries: { data: (number | undefined)[]; name: string }[] = [
        { data: [], name: 'weight' },
        { data: [], name: 'height' },
    ];
    // make sure events are sorted from oldest to latest
    const sortedSmsEvents = sortByEventDate(smsEvents).reverse();

    for (const data of sortedSmsEvents) {
        const calendarCategoryObj = {
            month: new Date(data.event_date).getMonth(),
            year: new Date(data.event_date).getFullYear(),
        };
        if (data.weight && data.height) {
            categories.push(`${getMonthNames(t)[calendarCategoryObj.month]}/${calendarCategoryObj.year}`);
            dataSeries[0].data.push(data.weight);
            dataSeries[1].data.push(data.height);
        }
    }
    return {
        categories,
        dataSeries,
    };
};

/** create a chart data series containing the weight data series from smsEvents */
export const getWeightDataSeries = (smsEvents: SmsData[], t: TFunction) => {
    const dataSeries = getWeightHeightDataSeries(smsEvents, t);
    const weightDataSeries = {
        categories: dataSeries.categories,
        dataSeries: [dataSeries.dataSeries[0]],
    };

    return weightDataSeries;
};

/** create a data series containing the diastolic and systolic data series from smsEvents */
export const getBloodPSeriesForChart = (smsEvents: SmsData[], t: TFunction) => {
    const categories: string[] = [];
    const dataSeries: { data: (number | undefined)[]; name: string }[] = [
        { data: [], name: 'systolic' },
        { data: [], name: 'diastolic' },
    ];
    // make sure events are sorted from oldest to latest
    const sortedSmsEvents = sortByEventDate(smsEvents).reverse();

    const regex = /\d{2,3}\S{0,3}\d{2,3}/;
    const singleBpValueRegex = /\d{2,3}/g;

    for (const data of sortedSmsEvents) {
        const calendarCategoryObj = {
            month: new Date(data.EventDate).getMonth(),
            year: new Date(data.EventDate).getFullYear(),
        };
        const hasValidBp = regex.test(data.bp);
        if (!hasValidBp) {
            continue;
        }
        categories.push(`${getMonthNames(t)[calendarCategoryObj.month]}/${calendarCategoryObj.year}`);
        const matchedValues = [...Array.from(data.bp.matchAll(singleBpValueRegex))];

        dataSeries[0].data.push(Number(matchedValues[0]));
        dataSeries[1].data.push(Number(matchedValues[1]));
    }
    return {
        categories,
        dataSeries,
    };
};

/** generate filter option filters from chunked smsEvents */
export const pregnancyOptionsFilter = (chunkedSms: SmsData[][], t: TFunction) => {
    return chunkedSms.map((_, index) => {
        const thisOption = {
            value: index,
            label: t(`pregnancy ${index + 1}`),
        };
        if (index === chunkedSms.length - 1) {
            thisOption.label = t(`Current Pregnancy`);
        }
        return thisOption;
    });
};
