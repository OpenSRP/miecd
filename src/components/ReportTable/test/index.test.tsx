import ReportTable from '..';
import { Dictionary } from '@onaio/utils';
import { mountWithTranslations } from 'helpers/testUtils';
import { createBrowserHistory } from 'history';
import React from 'react';
import store from 'store';
import toJson from 'enzyme-to-json';
import { Provider } from 'react-redux';
import { Router } from 'react-router';

jest.mock('highcharts');

const history = createBrowserHistory();
const locationProps = {
    history,
    location: {
        hash: '',
        pathname: `chart`,
        search: '',
        state: {},
    },
    match: {
        isExact: true,
        params: { patient_id: '1002KL' },
        path: `chart`,
        url: `chart`,
    },
};

jest.mock('react-select', () => ({ options, onChange }: Dictionary) => {
    function handleChange(event: Dictionary) {
        const option = options.find((option: Dictionary) => option.value === event.target?.value);
        onChange(option);
    }

    return (
        <select data-testid="select" onChange={handleChange}>
            {options.map(({ label, value }: Dictionary) => (
                <option key={value} value={value}>
                    {label}
                </option>
            ))}
        </select>
    );
});

describe('ReportTable', () => {
    beforeEach(() => {
        jest.resetAllMocks();
    });

    it('renders correctly', () => {
        const props = {
            ...locationProps,
        };
        const wrapper = mountWithTranslations(
            <Provider store={store}>
                <Router history={history}>
                    <ReportTable {...props} />
                </Router>
            </Provider>,
        );

        expect(toJson(wrapper.find('#filter-panel'))).toMatchSnapshot('Filter panel');
        expect(wrapper.find('#tableRow table')).toHaveLength(1);
        expect(wrapper.find('MotherChart')).toHaveLength(1);
        expect(wrapper.find('ChildChart')).toHaveLength(0);
    });
    it('renders child chart', () => {
        const props = {
            ...locationProps,
            isChild: true,
        };
        const wrapper = mountWithTranslations(
            <Provider store={store}>
                <Router history={history}>
                    <ReportTable {...props} />
                </Router>
            </Provider>,
        );

        expect(wrapper.find('MotherChart')).toHaveLength(0);
        expect(wrapper.find('ChildChart')).toHaveLength(1);
    });
});
