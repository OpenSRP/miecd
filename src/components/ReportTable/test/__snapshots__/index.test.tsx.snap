// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportTable works okay: after filter pregnancies 1`] = `"ReportDateReporterMessage"`;

exports[`ReportTable works okay: after filter pregnancies 2`] = `
"Pregnancy Registration2020-03-03Amanda BeckEDD = 28-06-2020
Gravidity = 4
Parity = 3
Weight = 78 
Height = 165
BMI = 28
Previous Gynaecology History = Home Delivery, Previous convulsion
Current Symptoms = Coughing"
`;

exports[`ReportTable works okay: initial current pregnancies 1`] = `"ReportDateReporterMessage"`;

exports[`ReportTable works okay: initial current pregnancies 2`] = `
"Pregnancy Registration2019-09-25Husna HarizEDD = 2019-04-01
Gravidity = 3
Parity = 2
Weight = 54 
Height = 160
Previous Gynaecology History = Diabetes, High blood pressure
Current Symptoms = Flaccid Paralysis"
`;

exports[`ReportTable works okay: initial current pregnancies 3`] = `
"Birth Report2019-09-25Husna HarizChild DOB = 2020-01-01 
Child Gender = M 
Delivery Location = District hospitals/CDC 
Newborn Symptoms = Hypothermia, Fever 
Child Weight = 25 
Child Length = 31 
Child Risk = red_alert 
Breastfeeding = Breastfeeding during 1st hour"
`;

exports[`ReportTable works okay: initial current pregnancies 4`] = `
"Birth Report2019-09-25Husna HarizChild DOB = 2020-04-01 
Child Gender = M 
Delivery Location = District hospitals/CDC 
Newborn Symptoms = Hypothermia, Fever 
Child Weight = 25 
Child Length = 31 
Child Risk = red_alert 
Breastfeeding = Breastfeeding during 1st hour"
`;

exports[`ReportTable works okay: initial current pregnancies 5`] = `
"Birth Report2019-09-27Husna HarizChild DOB = 2020-04-01 
Child Gender = M 
Delivery Location = District hospitals/CDC 
Newborn Symptoms = Hypothermia, Fever 
Child Weight = 25 
Child Length = 31 
Child Risk = red_alert 
Breastfeeding = Breastfeeding during 1st hour"
`;

exports[`ReportTable works okay: initial current pregnancies 6`] = `
"Birth Report2019-09-27Husna HarizChild DOB = 2020-04-01 
Child Gender = M 
Delivery Location = District hospitals/CDC 
Newborn Symptoms = Hypothermia, Fever 
Child Weight = 25 
Child Length = 31 
Child Risk = red_alert 
Breastfeeding = Breastfeeding during 1st hour"
`;

exports[`ReportTable works okay: initial current pregnancies 7`] = `
"Birth Report2019-09-27Husna HarizChild DOB = 2020-04-01 
Child Gender = M 
Delivery Location = District hospitals/CDC 
Newborn Symptoms = Hypothermia, Fever 
Child Weight = 25 
Child Length = 31 
Child Risk = red_alert 
Breastfeeding = Breastfeeding during 1st hour"
`;

exports[`ReportTable works okay: pregnancy filter 1`] = `
<select
  data-testid="select"
  onChange={[Function]}
>
  <option
    key="0"
    value={0}
  >
    pregnancy 0
  </option>
  <option
    key="1"
    value={1}
  >
    pregnancy 1
  </option>
  <option
    key="2"
    value={2}
  >
    Current Pregnancy
  </option>
</select>
`;
